src/test/java/Homework20.java

```diff
@@ -0,0 +1,41 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.devtools.v85.input.Input;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Wait;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import java.time.Duration;
+import java.util.List;
+
+
+public class Homework20 extends BaseTest {
+
+        @Test
+        public void deletePlayListTest() throws InterruptedException {
+            String playListName = "PlayListForDeleting";
+            login("demo@class.com", "te$t$tudent");
+            WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+            wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("#playlists i[role='button']")));
+            WebElement addPlayListButton = driver.findElement(By.cssSelector("#playlists i[role='button']"));
+            addPlayListButton.click();
+            wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[data-testid='playlist-context-menu-create-simple']")));
+            WebElement createPlayListButton = driver.findElement(By.cssSelector("[data-testid='playlist-context-menu-create-simple']"));
+            createPlayListButton.click();
+            WebElement playListNameInput = driver.findElement(By.cssSelector("[name='create-simple-playlist-form']>input"));
+            playListNameInput.sendKeys(playListName);
+            playListNameInput.sendKeys(Keys.ENTER);
+            wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//section[@id='playlists']//li/a[text()='PlayListForDeleting']")));
+            WebElement playList = driver.findElement(By.xpath("//section[@id='playlists']//li/a[text()='PlayListForDeleting']"));
+            playList.click();
+            wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[contains(@class, 'btn-delete-playlist')]")));
+            WebElement deleteButton = driver.findElement(By.xpath("//button[contains(@class, 'btn-delete-playlist')]"));
+            deleteButton.click();
+            wait.until(ExpectedConditions.invisibilityOfElementLocated(By.id("//section[@id='playlists']//li/a[text()='PlayListForDeleting']")));
+            Assert.assertTrue(wait.until(ExpectedConditions.invisibilityOfElementLocated(By.xpath("//section[@id='playlists']//li/a[text()='PlayListForDeleting']"))));
```

Comment:
If possible, avoid overly long methods. A long method is 20‒25 lines. If possible, you should split the method into several shorter ones or even create a new class for this.

For example

   **Use in test methods, the following logical split:   GIVEN – WHEN - THEN**
   
**Step 1**

![image](https://github.com/shavkatsalikhov/A52-QA-Automation-Framework/assets/104761482/b36818e3-1387-4f4e-97cb-25deed6531e0)

**Step 2**

```
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.testng.Assert;
import org.testng.annotations.Test;

public class Homework20 extends BaseTest {

    @Test
    public void deletePlayListTest() {
        // GIVEN
        String playListName = "PlayListForDeleting";
        login("demo@class.com", "te$t$tudent");

        // WHEN
        clickAddPlayListButton();
        clickCreatePlayListButton();
        enterPlayListName(playListName);
        pressEnterInPlayListNameInput();
        waitUntilPlayListIsVisible(playListName);
        openPlayList(playListName);
        clickDeleteButton();
        waitUntilPlayListIsInvisible(playListName);

        // THEN
        assertPlayListDeleted(playListName);
    }

    private void clickAddPlayListButton() {
        WebElement addPlayListButton = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("#playlists i[role='button']")));
        addPlayListButton.click();
    }

    private void clickCreatePlayListButton() {
        WebElement createPlayListButton = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[data-testid='playlist-context-menu-create-simple']")));
        createPlayListButton.click();
    }

    private void enterPlayListName(String playListName) {
        WebElement playListNameInput = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[name='create-simple-playlist-form']>input")));
        playListNameInput.sendKeys(playListName);
    }

    private void pressEnterInPlayListNameInput() {
        WebElement playListNameInput = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[name='create-simple-playlist-form']>input")));
        playListNameInput.sendKeys(Keys.ENTER);
    }

    private void waitUntilPlayListIsVisible(String playListName) {
        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//section[@id='playlists']//li/a[text()='" + playListName + "']")));
    }

    private void openPlayList(String playListName) {
        WebElement playList = wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//section[@id='playlists']//li/a[text()='" + playListName + "']")));
        playList.click();
    }

    private void clickDeleteButton() {
        WebElement deleteButton = wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[contains(@class, 'btn-delete-playlist')]")));
        deleteButton.click();
    }

    private void waitUntilPlayListIsInvisible(String playListName) {
        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.xpath("//section[@id='playlists']//li/a[text()='" + playListName + "']")));
    }

    private void assertPlayListDeleted(String playListName) {
        Assert.assertTrue(wait.until(ExpectedConditions.invisibilityOfElementLocated(By.xpath("//section[@id='playlists']//li/a[text()='" + playListName + "']"))));
    }
}
```

**General recommendations:**

Test status: 
![Test status_en](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/bfa74efc-b45b-4ca0-8d89-f5224a51bc66)

	
Test error (red). This status is assigned to tests that caused an exception from the tested source code.
Test error" means there's a problem with the code being tested. This is not okay for a tester. We need to fix it to make sure the software works correctly
![Test error_en](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/701a79a3-7c65-4a84-9180-e221c559ea02)

Test failed(yellow). If at least one test receives this status, then all its parents are marked as failed.
![Test failed_en](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/0171eacb-5e3f-4e2a-9a35-e0e626260d1d)

**Result Homework20:**

![image](https://github.com/shavkatsalikhov/A52-QA-Automation-Framework/assets/104761482/a60fd8cf-cc9a-4581-ac8f-113d493b1b16)